python/graphstorm/dataloading/dataset.py

```diff
@@ -354,780 +389,511 @@ def get_edge_feats(self, input_edges, edge_feat_field, device='cpu'):
 
-        It will setup self._test_idxs, the edge indices of the local test set.
-        The test_idxs can be empty.
+                train_idx = dgl.distributed.node_split(g.nodes[ntype].data[msk],
+                                                       pb, ntype=ntype, force_even=True,
+                                                       node_trainer_ids=node_trainer_ids)
+                assert train_idx is not None, "There is no training data."
+                num_train += len(train_idx)
+                train_idxs[ntype] = train_idx
 
-        Arguement
-        ---------
-        g: Dist DGLGraph
+                logging.debug('part %d | ntype %s, mask %s | train: %d',
+                              get_rank(), ntype, msk, len(train_idx))
+            else:
+                # Train mask may not exist for certain node types
+                logging.debug('part %d | ntype %s, mask %s | train: 0',
+                            get_rank(), ntype, msk)
+        logging.info('part %d, train %d', get_rank(), num_train)
+        return train_idxs
+
+    def _get_node_set(self, ntypes, mask):
+        """ called by get_node_val_set and get_node_test_set
         """
+        g = self._g
         pb = g.get_partition_book()
-        test_idxs = {}
-        infer_idxs = {}
-        # If eval_etypes is None, we use all edge types.
-        if self.eval_etypes is None:
-            self._eval_etypes = g.canonical_etypes
-        for canonical_etype in self.eval_etypes:
-            if 'test_mask' in g.edges[canonical_etype].data:
-                # test_mask exists
-                # we will do evaluation or inference on test data.
-                test_idx = dgl.distributed.edge_split(
-                    g.edges[canonical_etype].data['test_mask'],
-                    pb, etype=canonical_etype, force_even=True)
-                # If there are test data globally, we should add them to the dict.
-                if test_idx is not None and dist_sum(len(test_idx)) > 0:
-                    test_idxs[canonical_etype] = test_idx
-                    infer_idxs[canonical_etype] = test_idx
-            else:
-                # Inference only
-                # we will do inference on the entire edge set
-                if get_rank() == 0:
-                    logging.info("%s does not contains test_mask, skip testing %s. " + \
-                            "We will do inference on the entire edge set.",
-                            str(canonical_etype), str(canonical_etype))
-                infer_idx = dgl.distributed.edge_split(
-                    th.full((g.num_edges(canonical_etype),), True, dtype=th.bool),
-                    pb, etype=canonical_etype, force_even=True)
-                infer_idxs[canonical_etype] = infer_idx
+        idxs = {}
+        num_data = 0
+        ntypes = self._check_ntypes(ntypes)
+        masks = self._check_node_mask(ntypes, mask)
+
+        if isinstance(mask, str):
```

Comment:
Should the `self._check_node_mask()` do the same thing for the line #543 to #551?

Reply:
Good catch